### PR TITLE
feat: migrate public currency and exchange rate api to platform 

### PR DESCRIPTION
**Pull request by ravikumar-hash**
Clickup
https://app.clickup.com/t/86cwkequh

Description:
1. Added platformCommonApiService. 
2. Migrated public Currency and exchange rate Api's to platform
3. Added tests for services. 

## Summary by CodeRabbit

- **New Features**
	- Updated currency data with new entries and modifications, including the addition of 'Chinese Yuan Offshore' and updates to existing currency codes and names.
	- Introduced a new interface for structured exchange rate data representation.
	- Added a new service for improved API interactions.

- **Bug Fixes**
	- Adjusted API endpoints and query parameters for fetching exchange rates and currency lists to ensure accurate data retrieval.

- **Tests**
	- Updated test suites for `CurrencyService` and `RouterAuthService` to reflect changes in dependencies and ensure proper functionality.

### CODE REVIEW

File: src/app/core/services/currency.service.ts [hunk 1]
```diff
@@ -44,11 +48,13 @@ export class CurrencyService {
...
       switchMap((currentEou: ExtendedOrgUser) =>
-        this.apiService.get<CurrencyName>('/currency/all', {
-          params: {
-            org_id: currentEou && currentEou.ou && currentEou.ou.org_id,
-          },
-        })
+        this.platformCommonApiService
+          .get<PlatformApiResponse<CurrencyName>>('/currency/list', {
+            params: {
+              org_id: currentEou && currentEou.ou && currentEou.ou.org_id,
```
Comment (Dimple16) [inline src/app/core/services/currency.service.ts:54]: `org_id: currentEou?.ou?.org_id` this should work

File: src/app/core/services/currency.service.ts [hunk 2]
```diff
@@ -44,11 +48,13 @@ export class CurrencyService {
...
+            },
+          })
+          .pipe(map((response) => response.data))
```
Comment (Dimple16) [inline src/app/core/services/currency.service.ts:57]:
```
return from(this.authService.getEou()).pipe(
    switchMap((currentEou: ExtendedOrgUser) => {
      return this.platformCommonApiService.get<PlatformApiResponse<CurrencyName[]>>('/currency/list', {
        params: { org_id: currentEou?.ou?.org_id },
      });
    }),
    map((response) => response.data)
  );
```
I don't think another `pipe` is required here. Check if the above works

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/services/currency.service.ts (hunk 2 above)
Comment (ravikumar-hash) [reply src/app/core/services/currency.service.ts:57]: I tried  this method but it was not working, so i searched and found that, , map is applied after the switchMap, switchMap returns is an observable from the HTTP call, we want  map() to act on the response within the switchMap function. By placing map() inside the pipe(), we can correctly get the response data.

**Review by Dimple16 — APPROVED** [review]

**Review by Dimple16 — CHANGES_REQUESTED** [review]
@ravikumar-hash , can you check the unit tests?

File: src/app/core/services/platform-common-api.service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/internal/Observable';
+import { environment } from 'src/environments/environment';
```
[the remark this answers is not in the document]
Comment (Dimple16) [reply src/app/core/services/platform-common-api.service.ts:4]: @ravikumar-hash we should be using this `import { Observable } from 'rxjs';`

**Review by Dimple16 — APPROVED** [review]
added a comment

**Pull request merged**